src/map/packets/quest_mission_log.cpp

```diff
@@ -28,211 +28,123 @@
+                    packetType = EXP_MISS_CURRENT;
+                    break;
+                case LOG_MISS_COMP:
+                    switch (logID) {
+                    case MISSION_TOAU:
+                    case MISSION_WOTG:
+                        // Completed TOAU and WOTG missions share a packet
+                        generateCompleteExpMissionPacket(PChar);
+                        packetType = EXP_MISS_COMPLETE;
+                        break;
+                    case MISSION_ASSAULT:
+                        // Completed Assault Missions share a packet with completed TOAU quests
+                        generateQuestPacket(PChar, QUESTS_TOAU, LOG_QUEST_COMP);
+                        generateAssaultMissionPacket(PChar);
+                        packetType = ASSAULT_COMPLETE;
+                        break;
+                    case MISSION_CAMPAIGN:
+                        // Completed Campaign missions take up two packets. Second half will come in a follow-up packet.
+                        generateCampaignMissionPacket(PChar, 0);
+                        packetType = CMPGN_MISS_UN;
+                        break;
+                    }
+                    break;
+                case LOG_CAMPAIGN2:
+                    // Second Campaign packet, summoned through logType
+                    generateCampaignMissionPacket(PChar, 256);
+                    packetType = CMPGN_MISS_DEUX;
+                    break;
+            }
         }
-
-    case MISSION_ZILART:
-        if (status == 0x01) {
+        else {
+            // All other mission logs update current / completed with a standard "current mission" update
             generateCurrentMissionPacket(PChar);
-            logType = MISS_CURRENT;
-            break;
+            packetType = MISS_CURRENT;
         }
-        if (status == 0x02) {
-            generateCompleteMissionPacket(PChar);
-            logType = MISS_COMPLETE;
-            break;
+    }
+    else
+    {
+        // Now all that remains is TOAU quests.
+        generateQuestPacket(PChar, QUESTS_TOAU, logType);  // "Base" of the packet
+        if (logType == LOG_QUEST_CURR) {
+            // As before, current TOAU Quests share with TOAU Mission, WOTG, Assault, and Campaign Missions
+            generateCurrentExpMissionPacket(PChar); // Writes 12 bytes in same packet
         }
-    case MISSION_COP:
-    case MISSION_ACP:
-    case MISSION_AMK:
-    case MISSION_ASA:
-    case MISSION_SOA:
-    case MISSION_ROV:
-        if (status == 0x01)
-        {
-            generateCurrentMissionPacket(PChar);
-            logType = MISS_CURRENT;
-            break;
+        else {
+            // Completed TOAU Quests share a packet with completed Assault Missions
+            generateAssaultMissionPacket(PChar); // Writes in same packet
         }
+        packetType = QUEST_PACKET_BYTES.at(logID * 2 + logType);
```

Comment:
structure your map so you don't need to do this - unordered_map<int, pair>, maybe